packages/next-on-pages/src/buildApplication/processVercelFunctions/edgeFunctions.ts

```diff
@@ -0,0 +1,203 @@
+import { join } from 'node:path';
+import type { CollectedFunctions } from './configs';
+import { formatRoutePath, stripIndexRoute, validateFile } from '../../utils';
+import { cliWarn } from '../../cli';
+
+/**
+ * Processes the Edge function routes found in the Vercel build output.
+ *
+ * - Ensures middleware functions have use the correct file for the entrypoint.
+ * - Checks edge function entrypoints exists.
+ * - Tries to fix invalid functions that have a valid edge function that can be used instead.
+ *
+ * @param collectedFunctions Collected functions from the Vercel build output.
+ */
+export async function processEdgeFunctions(
+	collectedFunctions: CollectedFunctions,
+): Promise<void> {
+	const { edgeFunctions, invalidFunctions, ignoredFunctions } =
+		collectedFunctions;
+
+	const tempFunctionsMap = new Map<string, string>();
+	const foundRscFunctions = new Map<string, string>();
+
+	for (const [path, fnInfo] of edgeFunctions) {
+		const { result, finalEntrypoint } = await checkEntrypoint(
+			path,
+			fnInfo.relativePath,
+			fnInfo.config.entrypoint,
+		);
+
+		switch (result) {
+			case 'valid': {
+				// update the edge function with the final entrypoint if it changed
+				if (finalEntrypoint !== fnInfo.config.entrypoint) {
+					fnInfo.config.entrypoint = finalEntrypoint;
+				}
+
+				const formattedPathName = formatRoutePath(fnInfo.relativePath);
+
+				tempFunctionsMap.set(formattedPathName, path);
+				if (formattedPathName.endsWith('/index')) {
+					// strip `/index` from the path name as the build output config doesn't rewrite `/index` to `/`
+					tempFunctionsMap.set(stripIndexRoute(formattedPathName), path);
+				}
+
+				if (formattedPathName.endsWith('.rsc')) {
+					foundRscFunctions.set(formattedPathName, path);
+				}
+				break;
+			}
+			case 'invalid': {
+				invalidFunctions.set(path, fnInfo);
+				edgeFunctions.delete(path);
+				break;
+			}
+			case 'ignore': {
+				ignoredFunctions.set(path, fnInfo);
+				edgeFunctions.delete(path);
+				break;
+			}
+		}
+	}
+
+	// Remove any RSC functions that have a valid non-RSC function that can be used instead.
+	// NOTE: RSC functions are identical to non-RSC functions, so this is okay.
+	// https://github.com/vercel/vercel/blob/main/packages/next/src/server-build.ts#L1193
```

Comment:
did you mean this line?
```suggestion
	// https://github.com/vercel/vercel/blob/fdf86fda038133e50892fe6a28852ff2cdf56931/packages/next/src/server-build.ts#L1217
```

PS: generally I think it's safer to add the full commit/permanent link so that if things change later this comment doesn't get outdated (if the file were to update on main the line you are referring to could get out of sync and be irrelevant etc..., if they move or rename the file your link could even go to a 404 later on)

Reply:
I never tried that but it seems like permalinks using a shortened commit hash still work 🤔 🙂 
https://github.com/vercel/vercel/blob/fdf86f/packages/next/src/server-build.ts#L1217

Reply:
```suggestion
```suggestion
	// https://github.com/vercel/vercel/blob/fdf86f/packages/next/src/server-build.ts#L1217
```
```

Reply:
yeah... i just copied the same url we used before and, well, that proved your point perfectly... 😔